worker/uniter/runner/context/relation.go

```diff
@@ -175,3 +179,9 @@ func (ctx *ContextRelation) Suspended() bool {
 func (ctx *ContextRelation) SetStatus(status relation.Status) error {
 	return errors.Trace(ctx.ru.Relation().SetStatus(status))
 }
+
+// RemoteApplicationName returns the application on the other end of this
+// relation from the perspective of this unit.
+func (ctx *ContextRelation) RemoteApplicationName() string {
```

Comment:
Remote is making me think of CMR specifically here.  Can RequirerApplicationName be used here?

Reply:
My interpretation is that a relation has a local application (the one for which this unit is running a hook for) and the remote application on the other end. It's akin to the local and remote endpoints of a TCP socket.

However, if you think this is confusing, I could simply rename this to `OtherApplicationName` to be consistent with the uniter API method signatures (see above comment)

Reply:
Note that CMR or not makes no difference. You still have a local and remote application; the only difference is that the remote part can live in another model and/or controller.